### PR TITLE
[Enhancement][Lake] Avoid sending unnecessary publish version tasks

### PR DESCRIPTION
**Pull request by sduzh**
This patch is necessary for supporting schema changes on lake table.

For local tables(i.e., OlapTable), you can send PublishVersionRequest
to tablets of a shadow materialized index, even if the transaction does
not write any data to the shadow materialized index. But for LakeTable,
sending PublishVersionRequest to tablets of a shadow materialized index
that has not been written will result in errors.

This patch also removed two unused member variables(rollupIndexId and
rollupFinishedVersion) from MaterializedIndex.

Please note that the assignment to visibleTxnId is not included in this patch,
another patch will be submitted later to show how visibleTxnId is assigned.

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndex.java [hunk 1]
```diff
@@ -255,8 +246,8 @@ public void write(DataOutput out) throws IOException {
             tablet.write(out);
         }
 
-        out.writeLong(rollupIndexId);
-        out.writeLong(rollupFinishedVersion);
+        out.writeLong(-1L); // For rollback compatibility
+        out.writeLong(-1L); // For rollback compatibility
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndex.java]: Updated, PTAL

**Review by decster — DISMISSED** [review]

**Comment by sduzh** [comment]
@mergify rebase

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 23 (73.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|2|00.00%|[935, 936]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|3|00.00%|[311, 312, 313]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndex.java|17|18|94.44%|[309]|

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/load/Load.java [hunk 2]
```diff
@@ -908,7 +908,6 @@ public static void replayClearRollupInfo(ReplicaPersistInfo info, GlobalStateMgr
             OlapTable olapTable = (OlapTable) db.getTable(info.getTableId());
             Partition partition = olapTable.getPartition(info.getPartitionId());
             MaterializedIndex index = partition.getIndex(info.getIndexId());
-            index.clearRollupIndexInfo();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/load/Load.java:911]: This function can be removed.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/load/Load.java:911]: It has been removed.

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/load/Load.java (hunk 2 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/load/Load.java:911]: I mean `replayClearRollupInfo` can be removed.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/load/Load.java:911]: Removed, PTAL

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 23 (73.91%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|2|00.00%|[935, 936]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|3|00.00%|[311, 312, 313]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndex.java|17|18|94.44%|[309]|

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_admit_test

**Pull request merged**